js/src/forum/components/TextEditor.js

```diff
@@ -47,26 +53,28 @@ export default class TextEditor extends Component {
     );
   }
 
-  /**
-   * Configure the textarea element.
-   *
-   * @param {HTMLTextAreaElement} element
-   * @param {Boolean} isInitialized
-   */
-  configTextarea(element, isInitialized) {
-    if (isInitialized) return;
-
+  textareaOncreate(vnode) {
     const handler = () => {
       this.onsubmit();
       m.redraw();
     };
 
-    $(element).bind('keydown', 'meta+return', handler);
-    $(element).bind('keydown', 'ctrl+return', handler);
+    $(vnode.dom).bind('keydown', 'meta+return', handler);
+    $(vnode.dom).bind('keydown', 'ctrl+return', handler);
 
-    this.props.composer.editor = new SuperTextarea(element);
+    this.attrs.composer.editor = new SuperTextarea(vnode.dom);
   }
 
+  /**
+   * Here for extensibility
+   */
+  textareaOnupdate(vnode) {}
```

Comment:
Is this actually needed ? The `textarea` is a direct child of `TextEditor`. Extensions could simply extend `TextEditor.onupdate` and access `vnode.dom.querySelector('textarea')` from there. Since there are no conditionals  in `view` it's guaranteed to exist when the component's lifecycle events run.

We could also do the same above and it would save us one `bind`.

Reply:
Good point, done!